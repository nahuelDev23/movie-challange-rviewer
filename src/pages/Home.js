import React from 'react'
import { Categories } from '../components/Categories'

import { HomeLayout } from '../components/layout/HomeLayout'

export const Home = () => {
    return (
        <HomeLayout>
            <Categories />
        </HomeLayout>
    )
}
