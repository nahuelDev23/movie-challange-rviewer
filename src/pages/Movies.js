import React, { useEffect, useContext } from 'react'
import { Filter } from '../components/filter/Filter'
import { MainLayout } from '../components/layout/MainLayout'
import { DataContext } from '../context/data/DataContext'

export const Movies = () => {
  const { movies, onLoadData } = useContext(DataContext)

  useEffect(() => {
    onLoadData('movie')
  }, [])
  return (
    <MainLayout>
      <Filter category='movie' />
      <h1>movies</h1>
      {
        movies && movies.map(serie => (
          <div key={serie.title}>{serie.title}</div>
        ))
      }
    </MainLayout>
  )
}
