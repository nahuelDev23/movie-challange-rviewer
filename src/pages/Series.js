import React from 'react'
import { useContext } from 'react'
import { useEffect } from 'react'
import { Filter } from '../components/filter/Filter'
import { MainLayout } from '../components/layout/MainLayout'
import { DataContext } from '../context/data/DataContext'

export const Series = () => {

  const { series, onLoadData } = useContext(DataContext)

  useEffect(() => {
    onLoadData('series')
  }, [])

  return (
    <MainLayout>
      <Filter category='series' />
      <h1>Popular Series</h1>
      {
        series && series.map(serie => (
          <div key={serie.title}>{serie.title}</div>
        ))
      }
    </MainLayout>
  )
}
