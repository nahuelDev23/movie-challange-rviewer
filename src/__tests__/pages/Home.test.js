import { Home } from '../../pages/Home'
import { shallow } from 'enzyme'

describe('Pruebas en <Home/>', () => {

    test('Debe mostrar <Home/> correctamente', () => {
        const wrapper = shallow(<Home />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe hacer uso del HomeLayout', () => {
        const wrapper = shallow(<Home />);
        const HomeLayoutComponent = wrapper.find('HomeLayout');
        expect(HomeLayoutComponent).toBeTruthy();
    })
    test('Debe contener el componente Categories', () => {
        const wrapper = shallow(<Home />);
        const categoryComponent = wrapper.find('Categories');
        expect(categoryComponent).toBeTruthy();
    })

})