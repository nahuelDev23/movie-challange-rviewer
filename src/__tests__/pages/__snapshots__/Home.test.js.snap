// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Home/> Debe mostrar <Home/> correctamente 1`] = `
<HomeLayout>
  <Categories />
</HomeLayout>
`;
